prometheus/collector.go

```diff
@@ -33,7 +33,8 @@ type Collector interface {
 	// duplicate descriptors. Those duplicates are simply ignored. However,
 	// two different Collectors must not send duplicate descriptors.) This
 	// method idempotently sends the same descriptors throughout the
-	// lifetime of the Collector.
+	// lifetime of the Collector. A Collector unable to describe itself must
```

Comment:
"A Collector unable to describe itself" could be interpreted by a newbie as a Collector which generally doesn't support describing itself. Since such a thing is not supported, maybe write it like "If a Collector encounters an error while describing itself, it must send an invalid descriptor..."